CONTRIBUTING.md

```diff
@@ -0,0 +1,175 @@
+-   **Opening a discussion:** This repository comes with a discussions board where you can also ask for help. Click the "Discussions" tab at the top.
+
+As Foundry is still in heavy development, the documentation can be a bit scattered.
+The [Foundry Book][foundry-book] is our current best-effort attempt at keeping up-to-date information.
+
+### Submitting a bug report
+
+When filing a new bug report in the issue tracker, you will be presented with a basic form to fill out.
+
+If you believe that you have uncovered a bug, please fill out the form to the best of your ability. Do not worry if you cannot answer every detail; just fill in what you can. Contributors will ask follow-up questions if something is unclear.
+
+The most important pieces of information we need in a bug report are:
+
+-   The Foundry version you are on (and that it is up to date)
+-   The platform you are on (Windows, macOS, an M1 Mac or Linux)
+-   Code snippets if this is happening in relation to testing or building code
+-   Concrete steps to reproduce the bug
+
+In order to rule out the possibility of the bug being in your project, the code snippets should be as minimal
+as possible. It is better if you can reproduce the bug with a small snippet as opposed to an entire project!
+
+See [this guide][mcve] on how to create a minimal, complete, and verifiable example.
+
+### Submitting a feature request
+
+When adding a feature request in the issue tracker, you will be presented with a basic form to fill out.
+
+Please include as detailed of an explanation as possible of the feature you would like, adding additional context if necessary.
+
+If you have examples of other tools that have the feature you are requesting, please include them as well.
+
+### Resolving an issue
+
+Pull requests are the way concrete changes are made to the code, documentation, and dependencies of Foundry.
+
+Even minor pull requests, such as those fixing wording, are greatly appreciated. Before making a large change, it is usually
+a good idea to first open an issue describing the change to solicit feedback and guidance. This will increase
+the likelihood of the PR getting merged.
+
+Please also make sure that the following commands pass if you have changed the code:
+
+```sh
+forge fmt --check
+forge test -vvv
+```
+
+#### Adding cheatcodes
+
+Please follow the guide outlined in the [cheatcodes](https://github.com/foundry-rs/foundry/blob/master/docs/dev/cheatcodes.md#adding-a-new-cheatcode) documentation of Foundry.
+
+When making modifications to the native cheatcodes or adding new ones, please make sure to run [`./scripts/vm.py`](./scripts/vm.py) to update the cheatcodes in the [`src/Vm.sol`](./src/Vm.sol) file. This script will automatically generate the cheatcodes from the [`cheatcodes.json`](https://raw.githubusercontent.com/foundry-rs/foundry/master/crates/cheatcodes/assets/cheatcodes.json) file.
```

Comment:
Can we update this to mention the new arg in https://github.com/foundry-rs/forge-std/pull/619 since it's needed for some workflows, like @Tudmotu's?

Reply:
Done, documented the `--from` flag as well